modules/sources/transformers/snapshots-v3-transformer.ts

```diff
@@ -0,0 +1,124 @@
+import _ from 'lodash';
+import { PoolSnapshotFragment } from '../subgraphs/balancer-v3-vault/generated/types';
+import { Chain, PrismaPoolSnapshot, PrismaToken } from '@prisma/client';
+import { weiToFloat } from '../../common/numbers';
+
+/**
+ * Takes V3 subgraph snapshots and transforms them into DB entries.
+ *
+ * @param poolId - pool ID
+ * @param poolTokens - list of token addresses
+ * @param epoch - timestamp of the current day
+ * @param chain - chain
+ * @param allTokens - list of all tokens, used to get decimals
+ * @param prices - token prices used to calculate USD values
+ * @param previousDaySnapshot - previous day snapshot used to calculate daily values and acts as a fallback for total values when there were no swaps in the current day
+ * @param snapshot - V3 snapshot
+ * @returns
+ */
+export const snapshotsV3Transformer = async (
+    poolId: string,
+    poolTokens: string[],
+    epoch: number,
+    chain: Chain,
+    allTokens: PrismaToken[],
+    prices: Record<string, number>,
+    previousDaySnapshot?: PrismaPoolSnapshot,
+    snapshot?: PoolSnapshotFragment,
+): Promise<PrismaPoolSnapshot> => {
+    // Subgraph is storing balances in wei, we need to convert them to float using token decimals
+    const decimals = Object.fromEntries(
+        allTokens.filter((t) => poolTokens.includes(t.address)).map((t) => [t.address, t.decimals]),
+    );
+
+    // Use when the pool is new and there are no snapshots yet
+    const defaultZeros = Array.from({ length: poolTokens.length }, () => '0');
+
+    // `poolId-epoch` is used as the ID
+    const base = {
+        id: `${poolId}-${epoch}`,
+        chain,
+        poolId,
+        timestamp: epoch,
+    };
+
+    const values = {
+        totalShares: snapshot?.totalShares || previousDaySnapshot?.totalShares || '0',
+        totalSharesNum: weiToFloat(snapshot?.totalShares || previousDaySnapshot?.totalShares || '0', 18),
+        swapsCount: Number(snapshot?.swapsCount) || previousDaySnapshot?.swapsCount || 0,
+        holdersCount: Number(snapshot?.holdersCount) || previousDaySnapshot?.holdersCount || 0,
+        totalVolumes:
+            snapshot?.totalVolumes.map((balance, index) => {
+                const address = poolTokens[index];
+                return String(weiToFloat(balance, decimals[address] || 18));
```

Comment:
this could just be viem.formatUnits?

Reply:
formatUnits takes bigint only, so it would need to be:
```suggestion
                return formatUnits(BigInt(balance), decimals[address] || 18);
```

Reply:
looks better to me :)